runtime/gc_glue_java/ArrayletObjectModelBase.hpp

```diff
@@ -106,13 +106,24 @@ class GC_ArrayletObjectModelBase
 	MMINLINE UDATA
 	getSizeInElements(J9IndexableObject *arrayPtr)
 	{
-		UDATA size = ((J9IndexableObjectContiguous *)arrayPtr)->size;
+		UDATA size = 0;
+		if (compressObjectReferences()) {
+			size = ((J9IndexableObjectContiguousCompressed *)arrayPtr)->size;
 #if defined(J9VM_GC_HYBRID_ARRAYLETS)
-		if (0 == size) {
-			/* Discontiguous */
-			size =((J9IndexableObjectDiscontiguous *)arrayPtr)->size;
-		}
+			if (0 == size) {
+				/* Discontiguous */
+				size =((J9IndexableObjectDiscontiguousCompressed *)arrayPtr)->size;
```

Comment:
```suggestion
				size = ((J9IndexableObjectDiscontiguousCompressed *)arrayPtr)->size;
```

Reply:
This is a nitpick so feel free to ignore

Reply:
Fixed.